zebra-network/src/peer/handshake.rs

```diff
@@ -227,7 +418,12 @@ pub async fn negotiate_version(
         version: constants::CURRENT_VERSION,
         services: our_services,
         timestamp,
-        address_recv: (PeerServices::NODE_NETWORK, *addr),
+        address_recv: (
+            PeerServices::NODE_NETWORK,
+            connected_addr
+                .get_transient_addr()
+                .unwrap_or_else(|| *UNSPECIFIED_IPV4_ADDR),
+        ),
         // TODO: detect external address (#1893)
         address_from: (our_services, config.listen_addr),
```

Comment:
If we're making an isolated connection, send an all-zeroes "our services", so that peers treat us as a client.

This is a defence-in-depth security fix.

Reply:
It might also be nice to turn this substitution into a named method:
```rust
            connected_addr
                .get_transient_addr()
                .unwrap_or_else(|| ***UNSPECIFIED_IPV4_ADDR),
```

Reply:
Actually comments would be better.

Reply:
This is afc4a08d89db87006b08c8a1b3afcf741fabf85f, I ended up doing a refactor.